plugins/outputs/pulsar/pulsar.go

```diff
@@ -0,0 +1,378 @@
+	if producer, ok := p.producerCache[topic]; !ok {
+		producerOptions := pulsar.ProducerOptions{
+			Topic:               topic,
+			BatchingMaxMessages: metricBatchSize,
+		}
+		switch p.CompressionType {
+		case 0:
+			producerOptions.CompressionType = pulsar.NoCompression
+		case 1:
+			producerOptions.CompressionType = pulsar.LZ4
+		case 2:
+			producerOptions.CompressionType = pulsar.ZLib
+		case 4:
+			producerOptions.CompressionType = pulsar.ZSTD
+		default:
+			producerOptions.CompressionType = pulsar.NoCompression
+		}
+
+		switch p.HashingScheme {
+		case "Murmur3_32Hash":
+			producerOptions.HashingScheme = pulsar.Murmur3_32Hash
+		case "JavaStringHash":
+			producerOptions.HashingScheme = pulsar.JavaStringHash
+		default:
+			producerOptions.HashingScheme = pulsar.JavaStringHash
+		}
+		if p.MaxPendingMessages > 0 {
+			producerOptions.MaxPendingMessages = p.MaxPendingMessages
+		}
+		duration, err := time.ParseDuration(p.BatchingMaxPublishDelay)
+		if err != nil {
+			producerOptions.BatchingMaxPublishDelay = duration
+		}
+		producer, err := p.client.CreateProducer(producerOptions)
+		if err != nil {
+			return nil, err
+		} else {
+			p.Log.Infof("Created producer for topic %s", topic)
+			p.producerCache[topic] = producer
+			return producer, nil
+		}
+
+	} else {
+		p.Log.Debugf("Returning cached producer for topic %s", topic)
+		return producer, nil
+	}
+
+}
+func (p *Pulsar) SetSerializer(serializer serializers.Serializer) {
+	p.serializer = serializer
+}
+
+func (p *Pulsar) Connect() error {
+	p.producerCache = make(map[string]pulsar.Producer)
+	splits := strings.Split(p.Topic, "/")
+	p.tenantNameSpace = strings.Join(splits[0:4], "/")
+	clientOptions := pulsar.ClientOptions{
+		URL:                        p.URL,
+		TLSAllowInsecureConnection: p.TLSAllowInsecureConnection,
+		TLSTrustCertsFilePath:      p.TLSTrustCertsFilePath,
+		TLSValidateHostname:        p.TLSValidateHostname,
+	}
+	if p.ConnectionTimeout != "" {
+		connectionTimeout, err := time.ParseDuration(p.ConnectionTimeout)
+		if err != nil {
+			p.Log.Error("Invalid ConnectionTimeout. Setting it to 30s ", err.Error())
+			clientOptions.ConnectionTimeout = 30 * time.Second
+		} else {
+			clientOptions.ConnectionTimeout = connectionTimeout
+		}
+	}
+	if p.OperationTimeout != "" {
+		operationTimeout, err := time.ParseDuration(p.OperationTimeout)
+		if err != nil {
+			p.Log.Error("Invalid OperationTimeout. Setting it to 30s", err.Error())
+			clientOptions.OperationTimeout = 30 * time.Second
+		} else {
+			clientOptions.OperationTimeout = operationTimeout
+		}
+
+	}
+
+	if p.AuthProvider == "tls" {
+		if p.TLSCertificatePath != "" && p.TLSPrivateKeyPath != "" {
+			clientOptions.Authentication = pulsar.NewAuthenticationTLS(p.TLSCertificatePath, p.TLSPrivateKeyPath)
+		} else {
+			return fmt.Errorf("tls auth specified but tls_certificate_path or tls_private_key_path is miising")
+		}
+
```

Comment:
Please invert the logic
```suggestion
		if p.TLSCertificatePath == "" || p.TLSPrivateKeyPath == "" {
			return fmt.Errorf("tls auth specified but tls_certificate_path or tls_private_key_path is miising")
		}
		clientOptions.Authentication = pulsar.NewAuthenticationTLS(p.TLSCertificatePath, p.TLSPrivateKeyPath)

```